core/trino-main/src/main/java/io/trino/server/remotetask/DynamicFiltersFetcher.java

```diff
@@ -124,6 +128,7 @@ public synchronized void updateDynamicFiltersVersionAndFetchIfNecessary(long new
     private synchronized void stop()
```

Comment:
this method is not called normally. only in erronous situations. You cannot rely on it to be executed.

Reply:
@sopel39 @raunaqmorarka  Yeah - you are right. Do you have quick clue if there is natural place where we know that DynFilFetcher finished its work. The way it is called is kinda convoluted

Reply:
Yes:

* there is no pending request from `DynamicFilterFetcher`
* all places which could call `updateDynamicFiltersVersionAndFetchIfNecessary` are stopped (it's only `ContinuousTaskStatusFetcher` atm IIRC)

Reply:
PTAL now :)